src/libs/Navigation/AppNavigator/createCustomFullScreenNavigator/CustomFullScreenRouter.tsx

```diff
@@ -6,41 +6,49 @@ import type {FullScreenNavigatorRouterOptions} from './types';
 
 type StackState = StackNavigationState<ParamListBase> | PartialState<StackNavigationState<ParamListBase>>;
 
-const isAtLeastOneInState = (state: StackState, screenName: string): boolean => !!state.routes.find((route) => route.name === screenName);
+const isAtLeastOneInState = (state: StackState, screenName: string): boolean => state.routes.some((route) => route.name === screenName);
 
 function adaptStateIfNecessary(state: StackState) {
     const isNarrowLayout = getIsNarrowLayout();
+    const workspaceCentralPane = state.routes.at(-1);
+    const topmostWorkspaceCentralPaneRoute = workspaceCentralPane?.state?.routes[0];
+    // When a screen from the FullScreenNavigator is opened from the deeplink then params should be passed to SCREENS.WORKSPACE.INITIAL from the variable defined below.
+    const workspacesCentralPaneParams =
+        workspaceCentralPane?.params && 'params' in workspaceCentralPane.params ? (workspaceCentralPane.params.params as Record<string, string | undefined>) : undefined;
 
     // There should always be SETTINGS.ROOT screen in the state to make sure go back works properly if we deeplinkg to a subpage of settings.
-    if (!isAtLeastOneInState(state, SCREENS.SETTINGS.ROOT)) {
+    if (!isAtLeastOneInState(state, SCREENS.WORKSPACE.INITIAL)) {
         // @ts-expect-error Updating read only property
         // noinspection JSConstantReassignment
         state.stale = true; // eslint-disable-line
 
         // This is necessary for ts to narrow type down to PartialState.
         if (state.stale === true) {
             // Unshift the root screen to fill left pane.
-            state.routes.unshift({name: SCREENS.SETTINGS.ROOT});
+            state.routes.unshift({
+                name: SCREENS.WORKSPACE.INITIAL,
+                params: topmostWorkspaceCentralPaneRoute?.params ?? workspacesCentralPaneParams,
+            });
         }
     }
 
     // If the screen is wide, there should be at least two screens inside:
     // - SETINGS.ROOT to cover left pane.
```

Comment:
Right? Or how should this comment be updated?
```suggestion
    // - WORKSPACE.INITIAL to cover left pane.
```